otherlibs/dune-glob/test/dune_glob_tests.ml

```diff
@@ -0,0 +1,30 @@
+open! Stdune
+module Glob = Dune_glob.V1
+
+let printf = Printf.printf
+
+let test glob s =
+  printf "%S matches %S == %b" (Glob.to_string glob) s (Glob.test glob s)
+
+let%expect_test _ =
+  let glob = Glob.of_string "test" in
+  printf "%S" (Glob.to_string glob);
+  [%expect {| "test" |}]
+
+let%expect_test _ =
+  let glob = Glob.of_string "te*" in
+  test glob "test";
+  [%expect {| "te*" matches "test" == true |}];
+  test glob "t";
+  [%expect {| "te*" matches "t" == false |}];
+  test glob "te";
+  [%expect {| "te*" matches "te" == true |}]
+
+let%expect_test _ =
+  let glob = Glob.of_string "*st" in
+  test glob "test";
+  [%expect {| "*st" matches "test" == true |}];
+  test glob "t";
+  [%expect {| "*st" matches "t" == false |}];
+  test glob "st";
+  [%expect {| "*st" matches "st" == false |}]
```

Comment:
@diml no idea why this fails, but it seems like it was failing before my PR.

Reply:
It is a bit difficult to understand whether the test passes or not, could add  a `expect:bool` argument to `test` and make it print a loud error in case of mismatch?

That's actually a workflow we need to improve in `ppx_expect` BTW.

Reply:
Regarding the behaviour, agreed that it can be surprising but it actually matches the documentation:

> `*` matches any sequence of characters, except if it comes first in which
> case it matches any character that is not `.` followed by anything